source/slang/slang-check-decl.cpp

```diff
@@ -2040,11 +2040,22 @@ namespace Slang
             }
         }
 
-        if (auto parentDecl = as<AggTypeDecl>(getParentDecl(varDecl)))
+        std::optional<TypeTag> varTypeTags;
```

Comment:
It looks like this lambda is not necessary to be reference binding as `varTypeTags` is not used anywhere else?

Reply:
This logic was there because `getTypeTags` isn't needed for all control paths in the past and we just wanted to call it on-demand. But the new logic means it is required for most paths, so I changed it to just get and store it in a variable.